### PR TITLE
Documentation on Pulp's storage logic

### PR DESCRIPTION
**Pull request by ammaritiz**
Added an FAQ section to the User Guide to address
some pulp storage logic queries.

closes #2758
https://pulp.plan.io/issues/2758

### CODE REVIEW

**Comment by asmacdo** [comment]
I am not fond of documenting plugin things in the core pulp docs. Also, I don't think we should mention a model attribute `unit_key_fields` in our user docs. Instead, it might make more sense to say that each plugin defines its own uniqueness by its "natural key". It might be awkward to do this though. Some plugin docs have a nice place for this http://docs.pulpproject.org/plugins/pulp_python/reference/python-type.html but others don't. 

@dkliban, what do you think?

File: docs/user-guide/faq.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+**What happens when a source repository changes the checksum type that is
+published in the repository?**
+
+Some of the content types, especially RPM, supports many checksum types while
+publishing a repository. Since, this checksum is a part of the
+'unit_key_fields', it is possible for a repository to have multiple copies of
+the same content unit. Therefore, you need to be careful when changing the
+checksum type while publishing.
+
+**How Pulp keeps track of units that belong to a particular repository?**
```
Comment (dkliban) [inline docs/user-guide/faq.rst]: Each repository is stored as a document in the 'repos' MongoDB collection. Each content type is stored in a collection with a prefix of 'units_'. Relationships between content and repositories are stored in the 'repo_content_units' collection.

File: docs/user-guide/faq.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+Pulp uses MongoDB collections to keep track of units in a particular
+repository.
+
+**How symlinks are generated during a repository publish?**
```
Comment (dkliban) [inline docs/user-guide/faq.rst]: How are symlinks generated during a repository publish?

Pulp deduplicate content when possible. As a result all content units are stored in one place. Published content is actually a symlink to a content unit elsewhere on disk. When publishing a repository Pulp uses the relationships stored in the 'repo_content_units' for a particular repository to determine which symlinks need to be published.

File: docs/user-guide/faq.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
+FAQs
```
Comment (dkliban) [inline docs/user-guide/faq.rst]: let's rename this to 'Content storage FAQ'

**Comment by ammaritiz** [comment]
@dkliban Thanks for reviewing. I have pushed the necessary changes.

**Comment by ammaritiz** [comment]
@dkliban Are we good to merge this?

**Pull request merged**